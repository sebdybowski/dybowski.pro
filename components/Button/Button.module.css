.Button {
    padding: 1rem 1.5rem;
    margin: 2rem 2rem 0 0;
    background-color: transparent;
    color: var(--text-color);
    text-transform: uppercase;
    border: none;
    font-size: 2rem;
    font-family: var(--font-family);
    line-height: 2rem;
    transition: transform .2s;
}

.Button:hover {
    cursor: pointer;
    transform: scale(1.15);
}

.Button--outlined {
    border-radius: 4rem;
    border: 0.25rem solid var(--text-color);
}

.Button--outlined:hover {
    color: var(--primary-color);
    background-color: var(--text-color);
}

.Button__Icon {
    margin-right: 1rem;
}