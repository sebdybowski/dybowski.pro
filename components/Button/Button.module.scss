@use 'styles/variables.scss' as *;

.Button {
    padding: 1rem 1.5rem;
    margin: 2rem 2rem 0 0;
    background-color: transparent;
    color: $text-color-dark;
    text-transform: uppercase;
    border: none;
    font-size: 1.5rem;
    font-family: $main-font;
    font-weight: bold;
    line-height: 1.5rem;
    transition: transform .2s;

    @media screen and (max-width: ($breakpoint-md - 1)) {
        margin: 1rem 1rem;
    }
}

.Button:hover {
    cursor: pointer;
    transform: scale(1.1);
}

.Button--outlined {
    border-radius: 4rem;
    border: 0.25rem solid $text-color-light;
}

.Button--outlined:hover {
    color: $text-color-dark;
    background-color: $text-color-light;
}

.Button--inverted {
    color: $text-color-light;
    border-color: $text-color-light;
}

.Button--primary {
    color: $brand-color-primary;
}

.Button__Icon {
    margin-right: 1rem;
}

.Button__Text {}
