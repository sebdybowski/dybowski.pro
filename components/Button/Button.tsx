import React, { ButtonHTMLAttributes, FC, Fragment, ReactNode } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import clsx from 'clsx';

import styles from './Button.module.css';

import type { IconProp } from '@fortawesome/fontawesome-svg-core';

type ButtonProps = {
  children: ReactNode,
  icon?: IconProp,
  outline?: boolean,
  primary?: boolean
};

const Button: FC<ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>> = ({
  children, icon, type = 'button', outline = false, primary = false,
}) => (
  <button
    className={clsx(
      styles.Button,
      outline && styles['Button--outlined'],
      primary && styles['Button--primary'],
    )}
    type={type}
    data-testid="Button"
  >
    {
      icon ?
      <Fragment>
        <FontAwesomeIcon
          icon={icon}
          size='1x'
          className={styles.Button__Icon}
        />
        {children}
      </Fragment> :
        children
    }
  </button>
);

export default Button;
