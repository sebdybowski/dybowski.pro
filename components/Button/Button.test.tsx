import React from 'react';
import { render } from '@testing-library/react';
import { faArrowUpRightFromSquare } from '@fortawesome/free-solid-svg-icons';

import Button from './Button';

describe('<Button />', () => {
  test('it should render with the default props', () => {
    const { asFragment } = render(
      <Button>Button Label</Button>);
    
    expect(asFragment()).toMatchSnapshot();
  });

  test('it should render with the optional props', () => {
    const { asFragment } = render(
      <Button
        outline
        primary
        icon={faArrowUpRightFromSquare}
      >Button Label</Button>);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
