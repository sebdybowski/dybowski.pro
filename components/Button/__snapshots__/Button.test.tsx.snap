// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it should render 1`] = `
<DocumentFragment>
  <button
    class="Button"
    data-testid="Button"
    type="button"
  >
    Button Label
  </button>
</DocumentFragment>
`;
